tests/subset.rs

```diff
@@ -1,39 +1,52 @@
 #![deny(unused_must_use)]
-//! Integration tests of the Subset protocol.
-
-mod network;
+pub mod net;
 
 use std::collections::{BTreeMap, BTreeSet};
 use std::iter::once;
 use std::sync::Arc;
 
 use hbbft::subset::{Subset, SubsetOutput};
-use hbbft::NetworkInfo;
+use hbbft::ConsensusProtocol;
 
-use crate::network::{Adversary, MessageScheduler, NodeId, SilentAdversary, TestNetwork, TestNode};
+use crate::net::adversary::{Adversary, NodeOrderAdversary, ReorderingAdversary};
+use crate::net::{NetBuilder, NewNodeInfo, VirtualNet};
 
+type NodeId = u16;
 type ProposedValue = Vec<u8>;
 
-fn test_subset<A: Adversary<Subset<NodeId, u8>>>(
-    mut network: TestNetwork<A, Subset<NodeId, u8>>,
+fn test_subset<A>(
+    mut net: VirtualNet<Subset<NodeId, u8>, A>,
     inputs: &BTreeMap<NodeId, ProposedValue>,
-) {
-    let ids: Vec<NodeId> = network.nodes.keys().cloned().collect();
+) where
+    A: Adversary<Subset<NodeId, u8>>,
+{
+    let ids: Vec<NodeId> = net.nodes().map(|node| *node.id()).collect();
+
+    let mut rng = rand::thread_rng();
 
     for id in ids {
         if let Some(value) = inputs.get(&id) {
-            network.input(id, value.to_owned());
+            let _ = net.send_input(id, value.to_owned(), &mut rng);
         }
     }
 
-    // Terminate when all good nodes do.
-    while !network.nodes.values().all(TestNode::terminated) {
-        network.step();
+    // Handle messages until all good nodes have terminated.
+    while !net.nodes().all(|node| node.algorithm().terminated()) {
+        let _ = net.crank_expect(&mut rng);
     }
 
-    // Verify that all instances output the same set.
-    let observer: BTreeSet<_> = network.observer.outputs().iter().cloned().collect();
-    for node in network.nodes.values() {
+    // Get reference value
+    let observer: BTreeSet<_> = net
```

Comment:
Not sure whether node 0 is an observer? Otherwise I'd just call it `value`.

Are all the ported tests dropping the observer nodes?
For now that's fine, I think, but we should make a note to either add them back later or abandon the observer concept.

Reply:
Yes, I thought that naming was a bit odd when porting from the old tests, I renamed it to "expected_value" which makes a lot more sense.
The strategy of using the first node's output as reference value I took from the new net simulator's documentation, I make a note in the code to indicate that the value should be taken from an observer as soon as we have such a concept in the new simlator.